ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/parser/SqlTestLoader.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.parser;
+
+import com.google.common.collect.ImmutableList;
+import io.confluent.ksql.test.KsqlTestException;
+import io.confluent.ksql.test.loader.TestLoader;
+import io.confluent.ksql.test.model.TestLocation;
+import io.confluent.ksql.test.parser.SqlTestLoader.SqlTest;
+import io.confluent.ksql.test.parser.TestDirective.Type;
+import io.confluent.ksql.test.tools.Test;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * The {@code SqlTestLoader} loads the test files that should be run
+ * by the Ksql testing tool based on a path and optional filters.
+ */
+public class SqlTestLoader implements TestLoader<SqlTest> {
+
+  private final Predicate<SqlTest> shouldRun;
+  private final Path path;
+
+  public SqlTestLoader(final Path path) {
+    this(t -> true, path);
+  }
+
+  /**
+   * @param testFilter filters out which tests to run
+   * @param path       the top-level dir to load
+   */
+  public SqlTestLoader(final Predicate<SqlTest> testFilter, final Path path) {
+    this.shouldRun = Objects.requireNonNull(testFilter, "testFilter");
```

Comment:
How would the `shouldRun` be used? Will there be tests in a test file that one can indicate that they should be skipped? If so, should we add an example to test.sql?

Reply:
the idea was to use `shouldRun` to filter tests to run or not run based on some regex exp. I wasn't going to implement it in this PR quite yet but I figured I might as well set it up here